### DISCUSSION TITLE
API choice to integrate OTBR

### DISCUSSION THREAD

**Discussion by agners** [openthread/openthread#8487] [Q&A] [answered]
We currently consider which API to use to integrate the OTBR. From what I can tell we have essentially 3(/4) options:

- OpenThread Unix socket (very much geared towards CLI, command outputs are not very machine parse-able)
- D-Bus (local only, completeness?)
- Web-Server/REST (completeness?)

Are there additional options to consider? What is the most future proof API of the available options?

Best regards,
Stefan

**Comment by jwhui** [comment, marked as answer]
Of the three options, D-Bus is currently the most complete and one that we continue to maintain. While the OpenThread Unix domain socket is arguably more complete, it is very focused on CLI, as you mentioned.

**Reply by agners** [reply]
Ok, thanks for confirmation. Is the REST API still active/something the project would accept PR's for?

**Reply by jwhui** [reply]
Contributions for the REST API are welcome!
